src/services/category.ts

```diff
@@ -0,0 +1,12 @@
+'use server';
+import 'server-only';
+
+import { PrismaClient } from '@prisma/client';
+import { CategorySchema } from '@/types/generated';
+import { z } from 'zod';
+
+const prisma = new PrismaClient();
```

Comment:
We should only create one `PrismaClient` and use it everywhere.

https://www.prisma.io/docs/orm/prisma-client/setup-and-configuration/databases-connections#prismaclient-in-long-running-applications